MiniApp/Classes/core/Storage/MAManifestCache.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+import CommonCrypto
+
+internal class MAManifestCache {
+
+    typealias KeysDictionary = [String: CachedMetaData]
+    let keychainStore = MiniAppKeyChain(serviceName: .miniAppManifestCache)
+
+    func saveManifestInfo(forMiniApp appId: String, versionId: String, manifest: CachedMetaData) {
+        guard !appId.isEmpty && !versionId.isEmpty else {
+            return
+        }
+        var keysDic = retrieveAllManifestInfo()
+        if keysDic != nil {
+            keysDic?[appId + "/" + versionId] = manifest
+        } else {
+            keysDic = [appId + "/" + versionId: manifest]
+        }
+
+        if let keys = keysDic {
+            keychainStore.setInfoInKeyChain(keys: keys)
+        }
+    }
+
+    func getManifestInfo(forMiniApp appId: String, versionId: String) -> CachedMetaData? {
+        guard !appId.isEmpty && !versionId.isEmpty else {
+            return nil
+        }
+        guard let allKeys = retrieveAllManifestInfo(), let manifestInfo = allKeys[appId + "/" + versionId] as CachedMetaData? else {
+            return nil
+        }
+        return manifestInfo
+    }
+
+    /// Remove Key from the KeyChain
+    /// - Parameter keyId: Mini app ID
+    internal func removeKey(forMiniApp appId: String, versionId: String) {
```

Comment:
Currently no keys are ever being removed, right? So we could end up with many cached manifests that we don't need.

As mentioned in the comment above, I think we only need to cache a single manifest (the latest fetched), so probably this cache class can be simplified.